models/issue.go

```diff
@@ -612,6 +612,19 @@ func (issue *Issue) changeStatus(e *xorm.Session, doer *User, repo *Repository,
 	if issue.IsClosed == isClosed {
 		return nil
 	}
+
+	// Check for open dependencies
+	if isClosed { // only check if we're about to close an issue, otherwise reopening an issue would fail when there are unsatisfied dependencies
```

Comment:
We should not check whether there are open deps if `IsDependenciesEnabled` is false.